### PR TITLE
Polish of Expansion of Preferences Manager

### PR DESCRIPTION
**Pull request by KCeh**
## Summary

Preferences collectors and screens have be merged 🎉 

## Changes

- Introduction of filter in Collector interface (if similar use cases pop up in future we will separate interfaces - interface segregation principle)
- Removal of specific screen and logic for targeted prefs
- Fix for edge to edge and top bar scrolling on new prefs screen
- Fix for permission screen

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

[Screen_recording_20250306_212701.webm](https://github.com/user-attachments/assets/afe8b8c1-ef1e-42b7-bd04-127ff982ae7b)

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: sentinel/src/main/kotlin/com/infinum/sentinel/ui/main/preferences/PreferencesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    override fun onEvent(event: PreferencesEvent) =
+        when (event) {
+            is PreferencesEvent.Cached -> {
+                contract.launch(Unit)
+            }
+        }
+}
+
+internal fun Int.dpToPx(): Int = (this * Resources.getSystem().displayMetrics.density).toInt()
```
Comment (AsimRibo) [inline sentinel/src/main/kotlin/com/infinum/sentinel/ui/main/preferences/PreferencesFragment.kt:158]: I think we can move this to `Int` extension file.

File: sentinel/src/main/kotlin/com/infinum/sentinel/ui/main/preferences/PreferencesFragment.kt [hunk 2]
```diff
@@ -77,11 +77,15 @@ internal class PreferencesFragment : BaseChildFragment<PreferencesState, Prefere
     private fun initUi() {
         when (preferenceType) {
             TARGETED_PREFERENCES -> binding.apply {
+                container.setPadding(0, 0, 0, 104.dpToPx())
+                nestedScrollView.setPadding(0, 0, 0, 64.dpToPx())
```
Comment (AsimRibo) [inline sentinel/src/main/kotlin/com/infinum/sentinel/ui/main/preferences/PreferencesFragment.kt:78]: I think we can move 104 and 64 to resources and use
`resources.getDimensionPixelSize(R.dimen.container_bottom_padding)`. So there is no need for our conversion.

**Review by AsimRibo — APPROVED** [review]
Thanks for jumping in ❤️ . Looks good and works good. We can merge it and then test everything once again on the main PR. When we catch time in future, we can re-write collector interfaces the right way.
